web.rb

```diff
@@ -76,3 +94,49 @@ def log_info(message)
   status 200
   return {:intent => payment_intent.id, :secret => payment_intent.client_secret}.to_json
 end
+
+# This endpoint is used by the example apps to register a Reader
+# from Stripe.
+# https://stripe.com/docs/api/terminal/readers/create
+post '/register_reader' do
+  begin
+    reader = Stripe::Terminal::Reader.create(
+      :registration_code => params[:registration_code],
+      :label => params[:label]
+    )        
+  rescue Stripe::StripeError => e
+    status 402
+    return log_info("Error registering reader! #{e.message}")
+  end
+
+  log_info("Reader registered: #{reader.id}")
+  
+  status 200
+  return reader.to_json
+end
+
+# This endpoint is used to create a customer and save a card source to it.
+# https://stripe.com/docs/terminal/js/workflows#save-source
+post '/save_card_to_customer' do
+  begin
+    card_source = Stripe::Source.create(
+      :type => "card",
+      :card => {
+        :card_present_source => params[:card_present_source_id],
+      },
+    )
+
+    customer = Stripe::Customer.create(
```

Comment:
hmm, can use `Customer.list(:email => )` to upsert a customer here? 

it'd be fine this way, but not ideal since we'd fill up a user's test data with example customers...